alibi/explainers/ale.py

```diff
@@ -0,0 +1,479 @@
+# flake8: noqa E131
+import copy
+import math
+from itertools import count
+import numpy as np
+import pandas as pd
+from typing import Callable, List, Optional, Tuple, Union, TYPE_CHECKING, no_type_check
+
+from alibi.api.interfaces import Explainer, Explanation
+from alibi.api.defaults import DEFAULT_META_ALE, DEFAULT_DATA_ALE
+
+if TYPE_CHECKING:
+    import matplotlib.pyplot as plt
+
+
+class ALE(Explainer):
+
+    def __init__(self,
+                 predictor: Callable,
+                 feature_names: List[str] = None,
+                 target_names: List[str] = None) -> None:
+        """
+        Accumulated Local Effects for tabular datasets. Current implementation supports first order
+        feature effects of numerical features.
+
+        Parameters
+        ----------
+        predictor
+            A callable that takes in an NxF array as input and outputs an NxT array (N - number of
```

Comment:
Yea, probably stuff for another PR on updating docstrings. There are some that are out of date too (e.g. CF still say they return a `dict` instead of an `Explanation`).

Reply:
Opened an issue for this (#229), so resolving.